lib/client.js

```diff
@@ -0,0 +1,635 @@
+    assert.optionalNumber(opts.defaultMaxStreamingSizeMB,
+        'defaultMaxStreamingSizeMB');
+    assert.optionalNumber(opts.maxUtilizationPct, 'maxUtilizationPct');
+    assert.bool(opts.standalone, 'standalone');
+
+    this.log = opts.log || bunyan.createLogger({ name: 'storinfo' });
+
+    /*
+     * If we're not running in standalone mode, create the restify client.
+     */
+    if (!opts.standalone) {
+        assert.string(opts.url, 'url');
+        assert.object(opts.cueballOpts, 'cueballOpts');
+
+        opts.cueballOpts.log = this.log;
+        var clientOpts = {
+            url: opts.url,
+            agent: new cueball.HttpAgent(opts.cueballOpts),
+            log: this.log
+        };
+
+        this.http = restify.createJsonClient(clientOpts);
+    }
+    this.standalone = opts.standalone;
+
+    EventEmitter.call(this);
+
+    /*
+     * The dcSharkMap is an object that maps datacenter names to an array of
+     * sharks sorted by available storage capacity that are all at or below the
+     * storage utilization threshold for normal manta requests.
+     */
+    this.dcSharkMap = null;
+    /*
+     * The operatorDcSharkMap is an object that maps datacenter names to an
+     * array of sharks sorted by available storage capacity that are all at or
+     * below the storage utilization threshold for operator manta requests.
+     */
+    this.operatorDcSharkMap = null;
+
+    this.dcIndex = -1;
+    this.multiDC = opts.multiDC === undefined ? true : opts.multiDC;
+    this.datacenters = null;
+
+    if (opts.pollInterval !== undefined && !opts.standalone) {
+        this.pollInterval = (opts.pollInterval * 1000);
+        this.defMaxSizeMB = opts.defaultMaxStreamingSizeMB ||
+            DEF_MAX_STREAMING_SIZE_MB;
+        this.utilization = opts.maxUtilizationPct ||
+            DEF_MAX_PERCENT_UTIL;
+        setImmediate(doPoll.bind(this));
+    }
+}
+util.inherits(StorinfoClient, EventEmitter);
+
+StorinfoClient.prototype.close = function close() {
+    if (this.pollTimer) {
+        clearTimeout(this._storageTimer);
+    }
+    if (!this.standalone) {
+        this.http.close();
+    }
+};
+
+/*
+ * Get the data for a specific row (storage node) from the manta_storage bucket.
+ * This takes a single required argument, which is the manta_storage_id of the
+ * storage node.
+ *
+ * On success, the "err" parameter to the callback will be null and the "obj"
+ * param will contain manta_storage object for the requested storage node.
+ *
+ * On failure, the "obj" param will be set to null.
+ */
+StorinfoClient.prototype.pollSpecific = function pollSpecific(storageid,
+    callback) {
+
+    assert.string(storageid, 'storageid');
+
+    /*
+     * pollSpecific isn't supported in standalone mode
+     */
+    if (this.standalone) {
+        var standErr = new errors.StandaloneModeError();
+        callback(null, standErr);
+    }
+
+    this.http.get('/poll?only_id=' + storageid, function (err, req, res, obj) {
```

Comment:
See note below about using `this.http.get({path: ..., query: {only_id: storageid}, ...)`. That'll handle query param escaping properly. Granted that this client probably doesn't need to be defensive.

Reply:
Fixed.